app/server/lib/DocSnapshots.ts

```diff
@@ -350,15 +350,26 @@ export function shouldKeepSnapshots(snapshots: ObjSnapshotWithMetadata[], snapsh
 
   // Get time of current version
   const start = moment.tz(current.lastModified, tz);
-
-  // Track saved version per hour, day, week, month, year, and number of times a version
-  // has been saved based on a corresponding rule.
-  const buckets: TimeBucket[] = [
-    {range: 'hour', prev: start, usage: 0, cap: 25},
-    {range: 'day', prev: start, usage: 0, cap: 32},
-    {range: 'isoWeek', prev: start, usage: 0, cap: 12},
-    {range: 'month', prev: start, usage: 0, cap: 96},
-    {range: 'year', prev: start, usage: 0, cap: 1000}
+ // Read environment variable for cap values as a stringified JSON object with key-value pairs
+    const capObjectString = process.env.GRIST_SNAPSHOT_TIMEBUCKET_CAP || '{"hour": "25", "day": "32", "isoWeek": "12", "month": "96", "year": "1000"}';
+
+// Parse the stringified JSON object into an actual object
+    const caps = JSON.parse(capObjectString);
+
+// Extract the cap values for each bucket range and convert them to integers
+    const capHour = parseInt(caps.hour);
```

Comment:
There'd be no need for parsing if the values were passed as integers. You could also use `import { integerParam } from 'app/server/lib/requestUtils';` to check they are an integer and throw a meaningful error if not.

Reply:
thanks four suggestion. I'm implementing.